icu4c/source/common/unicode/uversion.h

```diff
@@ -133,12 +124,7 @@ typedef uint8_t UVersionInfo[U_MAX_VERSION_LENGTH];
 #   if U_USING_ICU_NAMESPACE
         U_NAMESPACE_USE
 #   endif
-#else
-#   define U_NAMESPACE_BEGIN
```

Comment:
I wonder if all usages of these macros are guarded by `U_SHOW_CPLUSPLUS_API` though?

Reply:
Yes, the test/hdrtst check exactly that.

As we discussed I will create a separate PR to guard all C++ headers, but as far as we fixed, these macros are now being used only by C++ code under the `__cplusplus` conditional (or the `U_SHOW_CPLUSPLUS_API` for that matter) or under the assumption that the global scope is C++.

Reply:
Thanks! :)